src/nvim/regexp.c

```diff
@@ -3000,9 +2999,9 @@ static int gethexchrs(int maxinputlen)
  * Get and return the value of the decimal string immediately after the
  * current position. Return -1 for invalid.  Consumes all digits.
  */
-static int getdecchrs(void)
+static int64_t getdecchrs(void)
 {
-  int nr = 0;
+  int64_t nr = 0;
```

Comment:
Vim used `long_u` here to cast it to `long` on return. I used `int64_t` to remove the need to cast. Alternative is `int32_t` and `uint32_t` to maintain the cast but that requires style changes to pass the lint build.